### PR TITLE
UI: Add ability to specify explicit virtualized list dimensions

### PR DESCRIPTION
**Pull request by roluk**
Begin steering towards getting rid of `react-virtualized-auto-sizer` package. This change adds optional `width` and `height` props to `VirtualizedPropertyGrid` and tree components.

Some reasons for switching to providing component dimensions explicitly:
* The measurement process used by `AutoSizer` isn't modern and thus more resource intensive than just a [`ResizeObserver`](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver).
* `AutoSizer` doesn't handle non-integer container heights well on Chrome browser, resulting in weird overflow behavior.
* It is a much more flexible solution which works with a larger variety of application layouts.
* Components with explicit sizes are mounted on the same React render cycle, making component `ref` accessible in the first `useEffect` call.
* After making the switch in unit tests, I have found a few features in our components that didn't have tests for them.
* Tests no longer require to mock container element sizes.

### CODE REVIEW

**Comment by roluk** [comment]
Found an issue when resizing tree component in presentation-test-app. Will fix before merging the PR.

File: ui/components/src/ui-components/common/ConditionalAutoSizer.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+/** @internal */
+export interface ConditionalAutoSizerProps {
+  width?: number;
+  height?: number;
+  onResize?: (newSize: Size) => void;
+  children: (size: Size) => React.ReactNode;
+}
+
+/** @internal */
+export function ConditionalAutoSizer(props: ConditionalAutoSizerProps): React.ReactElement {
```
Comment (bsteinbk) [inline ui/components/src/ui-components/common/ConditionalAutoSizer.tsx:17]: Even thought this is internal, could you add comment to this function describing its use?

Comment (roluk) [reply ui/components/src/ui-components/common/ConditionalAutoSizer.tsx:26]: Added some documentation in 30fd7b8e93fc04b93c9f9aea37caf789a9a2518e.

Comment (bsteinbk) [reply ui/components/src/ui-components/common/ConditionalAutoSizer.tsx:26]: Thanks!

**Review by bsteinbk — APPROVED** [review]

**Comment by roluk** [comment]
Found another issue with horizontal scrollbar inside a tree when width and height are specified through props. Will need to dig deeper to find a way to fix that without introducing any regressions.

**Review by grigasp — APPROVED** [review]

**Pull request merged**